#!/usr/bin/env python
PACKAGE = "drone_detector_tracking"

from dynamic_reconfigure.parameter_generator_catkin import *
import roslib
roslib.load_manifest(PACKAGE)

gen = ParameterGenerator()

gen.add("h_low",  int_t,    0, "H [HSV] lower limit" , 110,    0, 255)
gen.add("h_high", int_t,    0, "H [HSV] higher limit", 150,   0, 255)
gen.add("s_low",  int_t,    0, "S [HSV] lower limit" , 100,  0, 255)
gen.add("s_high", int_t,    0, "S [HSV] higher limit", 255,  0, 255)
gen.add("v_low",  int_t,    0, "V [HSV] lower limit" , 0,   0, 255)
gen.add("v_high", int_t,    0, "V [HSV] higher limit", 255,  0, 255)

gen.add("blur_size", int_t, 0, "blur kernel size (px)",   1, 0, 31)
gen.add("opening_size", int_t, 0, "opening kernel size (px)",   1, 0, 31)
gen.add("dilation_size", int_t, 0, "dilation kernel size (px)",   1, 0, 31)

gen.add("min_circularity", double_t, 0, "min circularity",   0.7, 0, 3)
gen.add("max_circularity", double_t, 0, "max circularity",   1.3, 0, 3)


exit(gen.generate(PACKAGE, "drone_detector_tracking", "color_detector_reconfigure"))